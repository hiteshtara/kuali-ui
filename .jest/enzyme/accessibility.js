/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
