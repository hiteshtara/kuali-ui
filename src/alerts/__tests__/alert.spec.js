/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import Alert from '..'

describe('Error Component', () => {
  it('should render an Error component with defaults', () => {
    const wrapper = shallow(<Alert />)
    expect(wrapper).toMatchSnapshot()
  })
})
