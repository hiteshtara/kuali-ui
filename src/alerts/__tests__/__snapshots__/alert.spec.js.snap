// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render an Error component with defaults 1`] = `
<CSSTransition
  appear={true}
  classNames="kuali-alert"
  in={true}
  mountOnEnter={true}
  timeout={
    Object {
      "appear": 650,
      "exit": 600,
    }
  }
  unmountOnExit={true}
>
  <div
    className="kuali-alert--container kuali-alert--type---info"
  >
    <div
      className="kuali-alert--icon---box"
    >
      <Icon
        aria-hidden={true}
        iconClassName="material-icons"
        name="info"
        variant="info"
      />
    </div>
    <div
      className="kuali-alert--data---box"
    >
      <h1>
        Info
      </h1>
      <p />
    </div>
    <div
      className="kuali-alert--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="clear"
          variant="default"
        />
      </IconButton>
    </div>
  </div>
</CSSTransition>
`;
